src/services/activities/activity_service.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  Activity,
+  default as ActivityMongoDAO,
+} from '../../dao/mongo/activities';
+import UserMongoDAO from '../../dao/mongo/users';
+import {
+  BaseProviderDAO,
+  ProviderActivity,
+  ProviderUser,
+} from '../../dao/providers/base';
+import { LoggerInstance } from '../../logger';
+import { isNullOrUndefined } from 'util';
+
+export class ActivityService {
```

Comment:
I like this pattern. I was trying to figure out a good way to model and share this logic.